src/string.cr

```diff
@@ -4166,6 +4166,35 @@ class String
     regex.matches? self, pos
   end
 
+  # Matches the regular expression *regex* against the entire string and returns
+  # the resulting `MatchData`.
+  # It also updates `$~` with the result.
+  #
+  # ```
+  # "foo".match_full(/foo/)   # => Regex::MatchData("foo")
+  # $~                        # => Regex::MatchData("foo")
+  # "fooo".match_full(/foo/)  # => nil
+  # $~                        # raises Exception
+  # ```
+  def match_full(regex : Regex) : Regex::MatchData?
+    match = /(?:#{regex})\z/.match_at_byte_index(self, 0, Regex::Options::ANCHORED)
```

Comment:
Very interesting. So, according to https://github.com/crystal-lang/crystal/pull/9312#issuecomment-635180570, the brackets can even be dropped.
```suggestion
    match = /#{regex}\z/.match_at_byte_index(self, 0, Regex::Options::ANCHORED)
```

Reply:
Yeah, I am aware that they're not necessary. I guess they ended up here as result of a copy and paste.